File: Backend/DietApp/Domain/Models/MealEntry.cs
using System;

namespace DietApp.Domain.Models
{
    public class MealEntry
    {
        public int ID { get; set; }
        public DateTime Date { get; set; }
        public int UserID { get; set; }
        public User User { get; set; }
        public string Name { get; set; }
        public Nutrients Nutrients { get; set; }
        public int WeightGram { get; set; }
        public float Kcal { get => WeightGram * Nutrients.Kcal / 100; }
        public float Carbs { get => WeightGram * Nutrients.Carbs / 100; }
        public float Fat { get => WeightGram * Nutrients.Fat / 100; }
        public float Protein { get => WeightGram * Nutrients.Protein / 100; }
    }
}

File: Mobile/DietApp/app/src/main/java/com/example/dietapp/models/entity/MealEntry.kt
package com.example.dietapp.models.entity

import androidx.room.ColumnInfo
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.dietapp.models.Nutrients
import java.util.*

@Entity(tableName = "meal_entry")
data class MealEntry(
    @PrimaryKey val id: Int,
    val name: String,
    val date: Date,
    @Embedded val nutrients: Nutrients,
    @ColumnInfo(name = "weight_gram") val weightGram: Int,
    val kcal: Double,
    val carbs: Double,
    val fat: Double,
    val protein: Double
)